Support indicating partition transform when add column to Iceberg table (#21206)
prestodb/presto: presto-docs/src/main/sphinx/connector/iceberg.rst

     ALTER TABLE iceberg.web.page_views DROP COLUMN location;

> When adding a column, we can indicate it as a partition column. To do that, we should indicate the ``transform`` functions for it,

Reviewer: ```suggestion
To add a new column as a partition column, identify the transform functions for the column.
```

"We" should be avoided in technical documentation. For some explanations of why, see the documentation recommended word list entries for 'we' in [GitLab](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#we), [Google](https://developers.google.com/style/word-list#we), and [Microsoft](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/we).

> and then the table would be partitioned by the transformed value of the column::

Reviewer: ```suggestion
The table is partitioned by the transformed value of the column::
```

     ALTER TABLE iceberg.web.page_views ADD COLUMN zipcode VARCHAR WITH (partitioning = 'identity');

     ALTER TABLE iceberg.web.page_views ADD COLUMN location VARCHAR WITH (partitioning = 'truncate(2)');

     ALTER TABLE iceberg.web.page_views ADD COLUMN location VARCHAR WITH (partitioning = 'bucket(8)');

.. note::

>     ``Day``, ``Month``, ``Year``, ``Hour`` partition column transform functions are not supported in Presto Iceberg connector yet.

Reviewer: ```suggestion
    ``Day``, ``Month``, ``Year``, ``Hour`` partition column transform functions are not supported in the Presto Iceberg connector.
```

Documentation should describe the current state of the product. "Yet" is an implied promise that this will, or is likely to happen that is out of scope for docs. Does that make sense?

Author: I think we would support these transform functions sooner or later. So how could we express such a meaning? I see in the previous paragraph `Partition Column Transform`, it also use 'yet' to imply this promise, should we keep consistent with it?

Reviewer: > I think we would support these transform functions sooner or later. So how could we express such a meaning? I see in the previous paragraph `Partition Column Transform`, it also use 'yet' to imply this promise, should we keep consistent with it?

The simple answer to your first question is in docs, we don't. Technical documentation is not the place to express such a meaning. For references, see the Google developer documentation style guide about [present tense](https://developers.google.com/style/tense), and [Document future features](https://developers.google.com/style/future). 

About your second question, you make a good point about the other use of "yet" on this page, in `Partition Column Transform`! Thank you for noticing it. I was focused on the new text you were adding. 

Yes, to be consistent, that other use of "yet" in `Partition Column Transform` should also be removed. 

I hope this helps, let me know!

Author: Thanks for your patient explanation. I have fixed it.

Reviewer: > Thanks for your patient explanation. I have fixed it, please take a look.

You're very welcome! It looks great.

     ALTER TABLE iceberg.web.page_views DROP COLUMN location;

When adding a column, we can indicate it as a partition column, we can as well use the ``transform`` functions and partition

> the table by the transformed value of the column::

Author: Thanks, fixed.

     ALTER TABLE iceberg.web.page_views ADD COLUMN zipcode VARCHAR WITH (partitioning = 'identity');

     ALTER TABLE iceberg.web.page_views ADD COLUMN location VARCHAR WITH (partitioning = 'truncate(2)');

     ALTER TABLE iceberg.web.page_views ADD COLUMN location VARCHAR WITH (partitioning = 'bucket(8)');

.. note::

    ``Day``, ``Month``, ``Year``, ``Hour`` partition column transform functions are not supported in Presto Iceberg

>         connector yet.

Author: Done.
